src/Symfony/Component/HttpKernel/Kernel.php

```diff
@@ -469,6 +463,32 @@ protected function initializeBundles()
     }
 
     /**
+     * @param BundleInterface $bundle
+     * @param bool $dependency
+     */
+    protected function addBundle($bundle, $dependency = false)
+    {
+        $name = $bundle->getName();
+
+        if (isset($this->bundles[$name]) && false === $dependency) {
```

Comment:
You're right but my proposal does not allow two different bundles with the same name to be registered! 

It something like this:
* If bundles have the **same name** and the **same FQCN**, just easily **continue**. (So if different bundles have the same exact dependency no exception is thrown)
* If bundles have the **same name** and the **different FQCN**, throw an **exception**.
* If bundles have the **different name** and the **different FQCN**, smile and **continue**. :)